### ISSUE TITLE
[helm3] unknown subcommand should return error

### ISSUE THREAD

**Issue by nauxliu** [helm/helm#6998]
The output of `helm version`:

```
version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

There is no error raised if I passed an unknown subcommand to Helm 3, no error message and the exit code is 0.
It's really hard to find out what is going wrong when I was debugging a helm 2 plugin. there supposed to be an error message to tell me which subcommand does not exist.

**Comment by marckhouzam** [comment]
I cannot reproduce this myself.
If I run, say, `helm3 bad`, I get the helm help message.
There is one scenario that I'm aware of where there is no error message, which is when your kubeconfig file is no longer valid.  See #6863.  But the error code is 1 in that case, which is not what you describe.
Can you provide your exact command?

**Comment by nauxliu** [comment]
@marckhouzam Yep, I  can get the help message, but it did not contain an error message like helm2, and the exit code is 0.

Helm 2  Error message:
```
Error: unknown command "bed" for "helm"
Run 'helm --help' for usage.
```

Reproduction Script:
```
$ docker run -it --rm ubuntu bash
$ apt update && apt install curl -y
$ curl -kL https://get.helm.sh/helm-v3.0.0-linux-amd64.tar.gz | tar zxv
$ ./linux-amd64/helm bed

The Kubernetes package manager

Common actions for Helm:

- helm search:    search for charts
- helm pull:      download a chart to your local directory to view
- helm install:   upload the chart to Kubernetes
- helm list:      list releases of charts

Environment variables:
...

$ echo $?
0
```

**Comment by marckhouzam** [comment]
Now I understand and you are correct.  I'm not sure if that was a conscious decision to immediately show the help.  I think that at a minimum, there should be and error return code. 

Would you like to try to make a PR for this?

**Comment by nauxliu** [comment]
I'm not sure I could find the cause because I'm not that familiar with Helm code and Golang. but I'll give it a try tomorrow.

**Comment by marckhouzam** [comment]
@nauxliu 

Thinking about this some more I realized that Cobra (which is used by Helm to define its CLI commands) would be involved in the behaviour you are seeing, which could complicate your investigation.  So I had a look myself and I am starting to believe you might have stumbled upon a debatable behaviour in Cobra itself.  

Basically, in Cobra, `command.execute(a [] string)` is used to run the command you typed.  That method decides that if you have passed a parameter (which is what an invalid subcommand is considered) to a command that requires a subcommand, it should print the help and not give an error code.

I need to think some more about it to see if anything reasonable can be done in Helm, or if it is Cobra that needs to be tweaked.

For reference, this is related to the change in #4070

**Comment by marckhouzam** [comment]
I believe we can fix the issue for the root helm command by removing this line:
https://github.com/helm/helm/blob/84de17e7e5255e5033967f005d4b4c05c024317c/cmd/helm/root.go#L354
This would cause
```
helm3 bad
```
to print the error message and give the error code. I believe this will address the concern about debugging plugins.

Scenarios such as `helm repo bad` or `helm plugin bad` would still show the help and no error code, but they behaved like this for helm v2 already.

@nauxliu is that a good enough solution for you?

**Comment by nauxliu** [comment]
@marckhouzam Sounds good, should I make a PR or you'll?

**Comment by marckhouzam** [comment]
If you'd like to try it out you can go ahead. If it's too much setup let me know and I'll do it.

**Comment by nauxliu** [comment]
@marckhouzam I've figured out why `require.NoArgs` is not working, Cobra will return help message before validating args if the command is not runnable(`command.Run` and `command.RunE` are both `nil`) https://github.com/spf13/cobra/blob/77e4d5aecc4d34e58f72e5a1c4a5a13ef55e6f44/command.go#L791-L804 

And Cobra already fixed this by changing to behavior to show full help, but do return the error if the command wasn't runnable, 
Ref: https://github.com/spf13/cobra/commit/9334a46bd6b3887f3561d705440038ec93b7f62e#diff-9c42b524cb14ca001c61267826cbefb1R792-R794

So I think the best way to fix this is by removing all `require.NoArgs` from unrunnable commands because they are not working, and upgrade cobra to the unreleased master. How do you think?

**Comment by nauxliu** [comment]
I was too naive, there is some inconsistency behavior in Cobra when handling Args and unknown commands, not easy to fix. I'll just remove the `require.NoArgs` from the root command.

**Comment by marckhouzam** [comment]
@nauxliu  good investigation! To be honest I was just as naive and hit the same Cobra behaviour in my investigation 😄 

I think the simple fix for the root command is sufficient. Thanks!
